### PR TITLE
PIM-3832: renamed servies name + specs

### PR DESCRIPTION
**Pull request by solivier**

### CODE REVIEW

File: src/Pim/Component/Connector/ArrayConverter/Flat/ProductToStandardConverter.php [hunk 1]
```diff
@@ -186,14 +186,22 @@ protected function convertToStructuredField($column, $value)
...
+                $attributeFieldInfo
+            ) && isset($attributeFieldInfo['attribute'])
+        ) {
+            $converter = $this->converterRegistry->getConverter($attributeFieldInfo['attribute']->getAttributeType());
 
-            if (null !== $converter) {
-                return $converter->convert($fieldNameInfo, $value);
+            if (null === $converter) {
+                throw new \LogicException(
+                    sprintf('No converters found for attribute type "%s"', $attributeFieldInfo['attribute']->getCode())
```
Comment (nidup) [inline src/Pim/Component/Connector/ArrayConverter/Flat/ProductToStandardConverter.php]: :+1:

File: src/Pim/Component/Connector/ArrayConverter/Flat/ProductToStandardConverter.php [hunk 2]
```diff
@@ -37,14 +37,14 @@ class ProductToStandardConverter implements StandardArrayConverterInterface
...
-     * @param ConverterRegistry               $converterRegistry
+     * @param ValueConverterRegistry               $converterRegistry
      * @param ProductAssociationFieldResolver $assocFieldResolver
      * @param FieldSplitter                   $fieldSplitter
      */
     public function __construct(
         ProductAttributeFieldExtractor $fieldExtractor,
         OptionsResolverConverter $optionsResolverConverter,
-        ConverterRegistry $converterRegistry,
+        ValueConverterRegistry $converterRegistry,
```
Comment (nidup) [inline src/Pim/Component/Connector/ArrayConverter/Flat/ProductToStandardConverter.php]: ValueConverterRegistryInterface if you have one :)

File: src/Pim/Component/Connector/ArrayConverter/Flat/Product/Converter/ValueConverterRegistryInterface.php [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+ * Registry of converters.
+ *
+ * @author    Olivier Soulet <olivier.soulet@akeneo.com>
+ * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+interface ValueConverterRegistryInterface
+{
+    /**
+     * Register a copier.
```
Comment (nidup) [inline src/Pim/Component/Connector/ArrayConverter/Flat/Product/Converter/ValueConverterRegistryInterface.php]: copier? ;)

File: src/Pim/Component/Connector/ArrayConverter/Flat/ProductToStandardConverter.php [hunk 4]
```diff
@@ -186,14 +186,22 @@ protected function convertToStructuredField($column, $value)
...
     {
-        $fieldNameInfo = $this->fieldExtractor->extractAttributeFieldNameInfos($column);
+        $attributeFieldInfo = $this->fieldExtractor->extractAttributeFieldNameInfos($column);
 
-        if (null !== $fieldNameInfo && array_key_exists('attribute', $fieldNameInfo) && isset($fieldNameInfo['attribute'])) {
-            $converter = $this->converterRegistry->getConverter($fieldNameInfo['attribute']->getAttributeType());
+        if (null !== $attributeFieldInfo && array_key_exists(
+                'attribute',
+                $attributeFieldInfo
+            ) && isset($attributeFieldInfo['attribute'])
```
Comment (skeleton) [inline src/Pim/Component/Connector/ArrayConverter/Flat/ProductToStandardConverter.php]: if (null !== $fieldNameInfo && isset($fieldNameInfo['attribute'])) is not enought ?

**Pull request merged**